### PR TITLE
HPCC4J-384 Support pushdown-filtering

### PR DESCRIPTION
**Pull request by rpastrana**
- Reports handled filters
- Integrates Spark filter in buidscan

Signed-off-by: Rodrigo Pastrana <Rodrigo.Pastrana@lexisnexisrisk.com>

### CODE REVIEW

File: DataAccess/src/test/java/org/hpccsystems/spark/HpccRelationTest.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        SparkConf sparkconf = new SparkConf();
+        sparkconf.setMaster(sparkMaster).setAppName("SPARKHPCCJUnit");
+        sparkcontext = new SparkContext(sparkconf);
+    }
+
+    @Test
+    @Category(org.hpccsystems.commons.annotations.RemoteTests.class)
+    public void testbuildScanAllValid() throws Exception
+    {
+        Assume.assumeTrue("Supported Spark Filters not available", supportedSparkFilters.size() > 0);
```
Comment (rpastrana) [inline DataAccess/src/test/java/org/hpccsystems/spark/HpccRelationTest.java:73]: @jpmcmu can you think of a way to test the buildscan functionality?

File: DataAccess/src/test/java/org/hpccsystems/spark/HpccRelationTest.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        supportedSparkFilters.add(new StringStartsWith("fixstr8", "Rod"));
+        supportedSparkFilters.add(new Or(new LessThan("int8", 12), new GreaterThan("field1", 8)));
+        supportedSparkFilters.add(new In("int8", new Object [] { "str", "values", "etc"}));
+        supportedSparkFilters.add(new In("int8", new Object [] { 1, 2, 3, 4, 5.6}));
+        supportedSparkFilters.add(new LessThan("fixstr8", "XYZ"));
+        supportedSparkFilters.add(new Not(new EqualTo("fixstr8", "true")));
+        supportedSparkFilters.add(new EqualTo("int8", 5));
+        supportedSparkFilters.add(new Not(new LessThan("int8", 3)));
+
+        unsupportedSparkFilters.add(new IsNull("something"));
```
Comment (jpmcmu) [inline DataAccess/src/test/java/org/hpccsystems/spark/HpccRelationTest.java:52]: It seems like there were more cases that we can't handle. Is that not the case?

**Review by jpmcmu — COMMENTED** [review]
@rpastrana looks good, only a few comments

File: DataAccess/src/test/java/org/hpccsystems/spark/HpccRelationTest.java (hunk 2 above)
Comment (rpastrana) [reply DataAccess/src/test/java/org/hpccsystems/spark/HpccRelationTest.java:52]: Surely there's many more cases, and we prob won't be able to cover them all here, but if you can think of any specific case let me know so we can add them.

**Comment by rpastrana** [comment]
@jpmcmu code review please.

**Review by jpmcmu — APPROVED** [review]
@rpastrana looks good

**Pull request merged**